### ISSUE TITLE
Generate CI filter for files included in `oxc_ast_codegen`

### ISSUE THREAD

**Issue by rzvxa** [oxc-project/oxc#4699]
Generate what files are watched, It is currently manually written here:

https://github.com/oxc-project/oxc/blob/82e2f6b92466fa7f22858a1d55be49e9ab4556f4/.github/workflows/ci.yml#L277-L284

**Comment by overlookmotel** [comment]
Can we just simplify it to:

```
filters: | 
  src: 
    - 'crates/oxc_ast/src/**'
    - 'crates/oxc_syntax/src/**'
    - 'crates/oxc_span/src/**'
    - 'tasks/ast_codegen/src/**' 
```

I know that casts the net a bit wider than it needs to, but:

1. It's foolproof.
2. These crates don't get that much churn (compared to e.g. `oxc_linter`, `oxc_transformer`), so job still wouldn't run on 95% of PRs.
3. Somehow generating a `ci.yml` file feels like overkill.

**Comment by rzvxa** [comment]
If I'm not mistaken filter can be its own file, So we only generate a simple `.github/ast_codegen_filter.yml`.
https://github.com/dorny/paths-filter?tab=readme-ov-file#advanced-options
It shouldn't be anything above a few lines of generator code(we just concat string instead of deserializing an actual yaml).

A simple generator not only makes sure that all correct paths are being watched but it would also add new paths to the list as we extend upon it. That's why I felt it would be a good way to set it up and just forget about it.

**Comment by overlookmotel** [comment]
Ah OK. If it's that easy then, yes, seems like a good idea.

**Comment by rzvxa** [comment]
> Ah OK. If it's that easy then, yes, seems like a good idea.

I'm optimistic that it is quite trivial, I hope time doesn't prove me wrong😆

**Comment by overlookmotel** [comment]
Implemented in #4737.
